src/app/services/results/results.service.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { SearchResultResponse } from '../../models/search-response.model';
+import { SearchItem } from '../../models/search-item.model';
+import { mockResponse } from '../../data/mock-response';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class ResultsService {
+    isResultsVisible = false;
+    private mockResponse: SearchResultResponse;
+    private mockItems: SearchItem[];
+    private searchResultsSubject = new BehaviorSubject<SearchItem[]>([]);
+    public searchResults = this.searchResultsSubject.asObservable();
+
+    constructor() {
+        this.mockResponse = mockResponse;
+        this.mockItems = mockResponse.items;
```

Comment:
please, don't use constructor for such cases) 
in angular constructor mostly used for declaring services and call super method like this 
 constructor(private resultsService: ResultsService) {}

for such purposes we use ngOnInit hook method

Reply:
```
    isResultsVisible = false;
    private mockResponse: SearchResultResponse = mockResponse;
    private mockItems: SearchItem[] = this.mockResponse.items;
    private searchResultsSubject = new BehaviorSubject<SearchItem[]>([]);
    public searchResults = this.searchResultsSubject.asObservable();
```
Fixed!